### PR TITLE
TST: df.loc[:, 'col'] returning a view, but df.loc[df.index, 'col'] returning a copy

### LINKED ISSUE #15631: df.loc[:, 'col'] returning a view, but df.loc[df.index, 'col'] returning a copy
#### Code Sample, a copy-pastable example if possible

```
x = pd.DataFrame(zip(range(4), range(4)), columns=['a', 'b'])
print(x)
   a  b
0  0  0
1  1  1
2  2  2
3  3  3

q = x.loc[:, 'a']
q += 2
print(x)
   a  b
0  2  0
1  3  1
2  4  2
3  5  3

x = pd.DataFrame(zip(range(4), range(4)), columns=['a', 'b'])
print(x)
   a  b
0  0  0
1  1  1
2  2  2
3  3  3

q = x.loc[x.index, 'a']
q += 2
print(x)
   a  b
0  0  0
1  1  1
2  2  2
3  3  3
```

```
```
#### Problem description

[df.loc[:, 'col'] returning a view, but df.loc[df.index, 'col'] returning a copy, intended? how can I make sure it is returning a copy?]

#### Expected Output
I thought .loc[] was always returning a copy

pd.show_versions()

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
pandas: 0.18.1
nose: 1.3.7
pip: 8.1.2
setuptools: 27.2.0
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.1
statsmodels: 0.6.1
xarray: None
IPython: 5.1.0
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: 1.1.0
tables: 3.2.2
numexpr: 2.6.1
matplotlib: 1.5.3
openpyxl: 2.3.2
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: 3.6.4
bs4: 4.5.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.42.0
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #15631
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by WillAyd** [comment]
I'm not sure about this - are we expecting that this loc call copies? @mroeschke @jreback

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -346,6 +346,16 @@ def test_loc_index(self):
         result = df.loc[pd.array(mask, dtype="boolean")]
         tm.assert_frame_equal(result, expected)
 
+    def test_loc_copy_vs_view(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:349]: hmm I am almost sure we have other tests about this, can you check and co-locate

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (arw2019) [reply pandas/tests/indexing/test_loc.py]: Yes you're right. This one
https://github.com/pandas-dev/pandas/blob/f5b2e5a9a2431165b1e8002219c5d3962d7bc920/pandas/tests/indexing/test_loc.py#L872
is very similar so moved it there

**Comment by erfannariman** [comment]
@arw2019 are you still working on this?

**Comment by arw2019** [comment]
@erfannariman yes I think this is ready to go, waiting for review

**Comment by jreback** [comment]
@WillAyd 

> I'm not sure about this - are we expecting that this loc call copies? @mroeschke @jreback 

yes this has been the case for a long time

**Review by jreback — APPROVED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
hmm can you add both examples from the OP.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**